Explorer/Assets/DCL/LOD/Components/SceneLODInfo.cs

```diff
@@ -1,55 +1,114 @@
+
+            if (loadedLODAmount == 1)
             {
-                CurrentVisibleLOD?.Release();
-                CurrentVisibleLOD = CurrentLOD;
+                if (CurrentLODLevelPromise == 0)
+                {
+                    lods[0].screenRelativeTransitionHeight = metadata.CullRelativeHeight;
+                    lods[1].screenRelativeTransitionHeight = metadata.CullRelativeHeight - 0.001f;
+                }
+                else
+                {
+                    lods[0].screenRelativeTransitionHeight = 1;
+                    lods[1].screenRelativeTransitionHeight = metadata.CullRelativeHeight;
+                }
+                //We need to recalculate the bounds only when the first LOD was loaded (hopefully they both have the same bounds)
+                //Ideally we would set it from the ABConverter
+                metadata.LodGroup.RecalculateBounds();
             }
+            else if (loadedLODAmount == 2)
+            {
+                lods[0].screenRelativeTransitionHeight = (1 - metadata.CullRelativeHeight) / 2 + metadata.CullRelativeHeight;
+                lods[1].screenRelativeTransitionHeight = metadata.CullRelativeHeight;
+            }
+
+            metadata.LodGroup.SetLODs(lods);
+            CurrentLODLevelPromise = byte.MaxValue;
         }
 
-        public void ResetToCurrentVisibleLOD()
+        private void CalculateCullRelativeHeight(Renderer[] lodRenderers, float defaultFOV, float defaultLodBias, int loadingDistance)
         {
-            CurrentLOD = CurrentVisibleLOD;
+            float distance = (loadingDistance - 1) * ParcelMathHelper.PARCEL_SIZE;
+            if (lodRenderers.Length > 0)
+            {
+                var mergedBounds = lodRenderers[0].bounds;
+
+                // Encapsulate the bounds of the remaining renderers
+                for (int i = 1; i < lodRenderers.Length; i++) { mergedBounds.Encapsulate(lodRenderers[i].bounds); }
+
+                metadata.CullRelativeHeight = Mathf.Clamp(CalculateScreenRelativeTransitionHeight(defaultFOV, defaultLodBias, distance, mergedBounds), 0.02f, 0.999f);
+            }
+        }
+
+        public float CalculateScreenRelativeTransitionHeight(float defaultFOV, float defaultLodBias, float distance, Bounds rendererBounds)
+        {
+            float objectSize = Mathf.Max(Mathf.Max(rendererBounds.extents.x, rendererBounds.extents.y), rendererBounds.extents.z) * defaultLodBias;
+            float halfFov = (defaultFOV / 2.0f) * Mathf.Deg2Rad;
+            float ScreenRelativeTransitionHeight = objectSize / ((distance + objectSize) * Mathf.Tan(halfFov));
+            return ScreenRelativeTransitionHeight;
         }
-    }
 
+        public bool HasLOD(byte lodForAcquisition)
+        {
+            return SceneLODInfoUtils.HasLODResult(metadata.SuccessfullLODs, lodForAcquisition) ||
+                   SceneLODInfoUtils.HasLODResult(metadata.FailedLODs, lodForAcquisition) ||
```

Comment:
That is correct, yes. Once failed, it will remain failed unless its removed from the cached by the memory full functionality. 
Only just then, it will be retried.